src/pybind/mgr/orchestrator.py

```diff
@@ -1026,12 +1025,29 @@ class PlacementSpec(object):
     """
     For APIs that need to specify a node subset
     """
-    def __init__(self, label=None, nodes=None):
+    def __init__(self, label=None, nodes=None, count=None):
         self.label = label
         if nodes:
-            self.nodes = [parse_host_specs(x, require_network=False) for x in nodes]
+            self.nodes = [parse_host_specs(x, require_network=False) for x in nodes if x]
         else:
             self.nodes = []
+        self.count = count if count else 1
+
+    def set_nodes(self, nodes):
+        # To backpopulate the .nodes attribute when using labels or count
+        # in the orchestrator backend.
+        self.nodes = nodes
+
+    @classmethod
+    def from_yaml(cls, data):
```

Comment:
```suggestion
    def from_dict(cls, data):
```
? 
We should also call `validate()` here.